### ISSUE TITLE
Update proto file and grpc packages

### ISSUE THREAD

**Issue by mhoeger** [Azure/azure-functions-python-worker#146]
We'll be updating the functions host to consume this breaking change (we have a naming conflict between package namespace and name of GRPC service): https://github.com/Azure/azure-functions-language-worker-protobuf/commit/a42a580e5211f89d5cc605a869a48425fccbd880

I'll update this issue with the latest version of the functions host that consumes the proto file changes when it is available!

**Comment by pragnagopa** [comment]
@elprans _ please udpate grpc packages to the latest version 1.12.0

**Comment by pragnagopa** [comment]
Please find the artificats here- [2.0.11879](https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-script-y8o14/build/2.0.11879/job/xmbu4p2g9c8a7utu/artifacts). This is a breaking change please address this ASAP as the functions runtime roll out will be starting next week.

**Comment by pragnagopa** [comment]
@elprans - If you are already made progress on splitting the repos in dev branch, please submit fix for this issue in master branch.

**Comment by elprans** [comment]
@pragnagopa  I'm having issues with the new host build:

```
It was not possible to find any compatible framework version
The specified framework 'Microsoft.AspNetCore.App', version '2.1.0' was not found.
  - Check application dependencies and target a framework version installed at:
      /
  - Alternatively, install the framework version '2.1.0'.
```

Previous build works fine.  I have dotnetcore-sdk version 2.1.105 installed.

**Comment by pragnagopa** [comment]
can you please install the latest from https://www.microsoft.com/net/download/windows
We need 2.1.300

**Comment by elprans** [comment]
I'm on linux, and adter installing 2.1.300 I get:

```
Unhandled Exception: System.BadImageFormatException: An attempt was made to load a program with an incorrect format.
 (Exception from HRESULT: 0x8007000B)
```

**Comment by elprans** [comment]
Here's the installation info:
```
$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     gentoo
 OS Version:
 OS Platform: Linux
 RID:         gentoo-x64
 Base Path:   /opt/dotnet_core/sdk/2.1.300/

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.300 [/opt/dotnet_core/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [/opt/dotnet_core/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [/opt/dotnet_core/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.0 [/opt/dotnet_core/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by pragnagopa** [comment]
@fabiocav - Any ideas?

**Comment by fabiocav** [comment]
@elprans which one of the ZIP files are you using? You want to grab one of the `no-runtime` bits.

**Comment by elprans** [comment]
@fabiocav OK, the `no-runtime` build seems to run, but it doesn't register any functions:

```
tests/test_cosmosdb_functions.py Hosting environment: Production
Content root path: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions
Now listening on: http://[::]:43743
info: Host.Startup[0]
      Reading host configuration file '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/host.json'
info: Host.Startup[0]
      Host configuration file read:
      {
        "functionTimeout": "00:05:00",
        "http": {
          "routePrefix": "api",
          "maxConcurrentRequests": 5,
          "maxOutstandingRequests": 30
        },
        "queues": {
          "visibilityTimeout": "00:00:10"
        },
        "eventHub": {
          "maxBatchSize": 1000,
          "prefetchCount": 1000,
          "batchCheckpointFrequency": 1
        },
        "logger": {
          "defaultLevel": "Trace",
          "categoryLevels": {
            "Worker": "Trace"
          }
        },
        "healthMonitor": {
          "enabled": true,
          "healthCheckInterval": "00:00:10",
          "healthCheckWindow": "00:02:00",
          "healthCheckThreshold": 6,
          "counterThreshold": 0.8
        }
      }
info: Host.General[0]
      Starting Host (HostId=hammer-1643887886, InstanceId=54fc0803-4aa9-40fd-ad25-069cff6779c8, Version=2.0.11879.0, ProcessId=12552, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
info: Host.Startup[0]
      Loading custom extension 'EventGridExtensionConfig'
info: Host.Startup[0]
      Loaded custom extension: EventGridExtensionConfig from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.Extensions.EventGrid.dll'
info: Host.Startup[0]
      Loading custom extension 'EventHubConfiguration'
info: Host.Startup[0]
      Loaded custom extension: EventHubConfiguration from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.EventHubs.dll'
info: Host.Startup[0]
      Loading custom extension 'CosmosDBConfiguration'
info: Host.Startup[0]
      Loaded custom extension: CosmosDBConfiguration from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.Extensions.CosmosDB.dll'
info: Host.Startup[0]
      Generating 0 job function(s)
warn: Host.Startup[0]
      No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).
```

**Comment by fabiocav** [comment]
I'm assuming what you have are Python functions, so I'll let @pragnagopa continue to assist as this is a different issue. Now that you have the runtime running, she should be able to help understand why those functions are not being loaded.

**Comment by pragnagopa** [comment]
Yes. I will look into this.

**Comment by pragnagopa** [comment]
@elprans - I cannot repro this issue. None of the functions are being indexed? or just the ones that need extensions?

**Comment by elprans** [comment]
None of the functions get indexed on my end with https://ci.appveyor.com/api/buildjobs/xmbu4p2g9c8a7utu/artifacts/Functions.Binaries.2.0.11879-alpha.no-runtime.zip

**Comment by pragnagopa** [comment]
@elprans - Please enable verbose logs by adding following in the host.json
```
"logger": {
    "categoryFilter": {
      "defaultLevel": "Trace"
    },
    "fileLoggingMode": "always"
  }
```

**Comment by elprans** [comment]
Here are the logs with trace level enaled:

```
$ env PYAZURE_WEBHOST_DEBUG=1 pytest -k test_cosmosdb
========================================================================================= test session starts =========================================================================================
platform linux -- Python 3.6.5, pytest-3.6.1, py-1.5.3, pluggy-0.6.0
rootdir: /home/elvis/dev/clients/microsoft/azure-functions-python-worker, inifile: setup.cfg
collected 62 items / 60 deselected                                                                                                                                                                    

tests/test_cosmosdb_functions.py Hosting environment: Production
Content root path: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions
Now listening on: http://[::]:46797
trce: Host.Startup[0]
      Host configuration applied.
info: Host.Startup[0]
      Reading host configuration file '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/host.json'
info: Host.Startup[0]
      Host configuration file read:
      {
        "functionTimeout": "00:05:00",
        "http": {
          "routePrefix": "api",
          "maxConcurrentRequests": 5,
          "maxOutstandingRequests": 30
        },
        "queues": {
          "visibilityTimeout": "00:00:10"
        },
        "eventHub": {
          "maxBatchSize": 1000,
          "prefetchCount": 1000,
          "batchCheckpointFrequency": 1
        },
        "logger": {
          "fileLoggingMode": "always",
          "categoryFilter": {
            "Worker": "Trace",
            "defaultLevel": "Trace"
          }
        },
        "healthMonitor": {
          "enabled": true,
          "healthCheckInterval": "00:00:10",
          "healthCheckWindow": "00:02:00",
          "healthCheckThreshold": 6,
          "counterThreshold": 0.8
        }
      }
info: Host.General[0]
      Starting Host (HostId=hammer-1643887886, InstanceId=c59b678e-dd57-4f37-979a-80fbd1f261b8, Version=2.0.11879.0, ProcessId=17582, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
trce: Host.Startup[0]
      Debug file watch initialized.
trce: Host.Startup[0]
      File event source initialized.
trce: Host.Startup[0]
      Workers Directory set to: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/build/webhost/workers
trce: Host.Startup[0]
      Loading worker providers from the workers directory: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/build/webhost/workers
trce: Host.Startup[0]
      Found worker config: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/build/webhost/workers/node/worker.config.json
trce: Host.Startup[0]
      Will load worker provider for language: node
trce: Host.Startup[0]
      Worker path for language worker java: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/build/webhost/workers/java
trce: Host.Startup[0]
      Worker path for language worker node: /home/elvis/dev/clients/microsoft/azure-functions-python-worker/build/webhost/workers/node
trce: Host.Startup[0]
      Function metadata read.
trce: Host.Startup[0]
      Binding providers loaded.
trce: Host.Startup[0]
      Binding providers initialized.
info: Host.Startup[0]
      Loading custom extension 'EventGridExtensionConfig'
info: Host.Startup[0]
      Loaded custom extension: EventGridExtensionConfig from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.Extensions.EventGrid.dll'
info: Host.Startup[0]
      Loading custom extension 'EventHubConfiguration'
info: Host.Startup[0]
      Loaded custom extension: EventHubConfiguration from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.EventHubs.dll'
info: Host.Startup[0]
      Loading custom extension 'CosmosDBConfiguration'
info: Host.Startup[0]
      Loaded custom extension: CosmosDBConfiguration from '/home/elvis/dev/clients/microsoft/azure-functions-python-worker/tests/cosmosdb_functions/bin/Microsoft.Azure.WebJobs.Extensions.CosmosDB.dll'
trce: Host.Startup[0]
      Extension loading complete.
trce: Host.Startup[0]
      Metadata provider created.
trce: Host.Startup[0]
      Adding Function descriptor providers for all languages.
trce: Host.Startup[0]
      Function descriptors created.
info: Host.Startup[0]
      Generating 0 job function(s)
warn: Host.Startup[0]
      No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).
info: Host.Startup[0]
      Host initialized (338ms)
info: Host.Startup[0]
      Host started (955ms)
info: Host.Startup[0]
      Job host started
info: Host.General[0]
      Host lock lease acquired by instance ID '00000000000000000000000036AAB76E'.
```

I think the host doesn't find the Python worker.  Has the configuration protocol changed.  Currently, we set the `workers:config:path` and `workers:python:path` environment variable.  The first points to the root, and the second points to `${ROOT}/python/worker.py`

**Comment by elprans** [comment]
Yes, I see the `Workers Directory set to` line is incorrect.

**Comment by pragnagopa** [comment]
Yes. Correct. Sorry missed mentioning that. You need to set languageWorkers:python:workerDirectory

**Comment by pragnagopa** [comment]
Here is the build for the current functions host release [2.0.11886](https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-script-y8o14/build/2.0.11886/job/0exdyn0sak341kea/artifacts)

**Comment by asavaritayal** [comment]
Available in release -  [1.0.300-alpha](https://github.com/Azure/azure-functions-python-worker/releases/tag/1.0.300-alpha)
